projects/RabbitMQ.Client/client/events/AsyncEventHandler.cs

```diff
@@ -1,21 +1,6 @@
-using System;
 using System.Threading.Tasks;
 
 namespace RabbitMQ.Client.Events
 {
-    public delegate Task AsyncEventHandler<in TEvent>(object sender, TEvent @event) where TEvent : EventArgs;
```

Comment:
This is a public API change, but it slips through the APIApproval test. 🙄 

(I removed it as the EventHandler<T> doesn't have it either (used to have it before 4.0 I think))
Also removing should be "fine" as it only makes it less strict => no recompilation errors AFAIK

Reply:
Since this is for 7.0, we can certainly make this change.

Reply:
If we want to backport it to 6.x and this is an issue, one could add it back (and to AsyncEventingWrapper) for 6.x